src/AuthenticatedAPIClient/axiosConfig.js

```diff
@@ -76,31 +76,52 @@ function applyAxiosInterceptors(authenticatedAPIClient) {
       authenticatedAPIClient.refreshAccessToken()
         .then(() => {
           queueRequests = false;
-          PubSub.publishSync(ACCESS_TOKEN_REFRESH);
+          PubSub.publishSync(ACCESS_TOKEN_REFRESH, { success: true });
         })
-        .catch(() => {
-          // TODO: We should give the client app an opportunity to
-          // take control here before logout/redirect to sign in.
-          authenticatedAPIClient.logout();
+        .catch((error) => {
+          if (authenticatedAPIClient.handleRefreshAccessTokenFailure) {
+            authenticatedAPIClient.handleRefreshAccessTokenFailure(error);
+          } else {
+            authenticatedAPIClient.logout();
+          }
+          PubSub.publishSync(ACCESS_TOKEN_REFRESH, { success: false });
         });
     }
 
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       logInfo(`Queuing API request ${originalRequest.url} while access token is refreshed`);
-      PubSub.subscribeOnce(ACCESS_TOKEN_REFRESH, () => {
-        logInfo(`Resolving queued API request ${originalRequest.url}`);
-        resolve(originalRequest);
+      PubSub.subscribeOnce(ACCESS_TOKEN_REFRESH, (msg, { success }) => {
+        if (success) {
+          logInfo(`Resolving queued API request ${originalRequest.url}`);
+          resolve(originalRequest);
+        } else {
+          reject(originalRequest);
+        }
       });
     });
   }
 
-  // Redirect to the logout page if an unauthorized API response was received.
+  // Log errors and info for unauthorized API responses
   function handleUnauthorizedAPIResponse(error) {
-    const errorStatus = error && error.response && error.response.status;
-    if (errorStatus === 401 || errorStatus === 403) {
-      logAPIErrorResponse(error, { errorFunctionName: 'handleUnauthorizedAPIResponse' });
-      authenticatedAPIClient.logout(authenticatedAPIClient.appBaseUrl);
+    const response = error && error.response;
+    const errorStatus = response && response.status;
+    const requestUrl = response && response.config && response.config.url;
+    const requestIsTokenRefresh = requestUrl === authenticatedAPIClient.refreshAccessTokenEndpoint;
+
+    switch (errorStatus) {
+      case 401:
+        if (requestIsTokenRefresh) {
+          logInfo(`Unauthorized token refresh response from ${requestUrl}`);
```

Comment:
Can we add to this message so when we see these it doesn't seem like something is going wrong?
```
Unauthorized token refresh response from ${requestUrl}. This is expected if the user is not yet logged in.
```

Reply:
Also, coverage is stating that your tests aren't covering all of these different cases.  I would either cover them, or add a pragma to state that you explicitly don't care.